lib/util/object_utils.js

```diff
@@ -94,6 +94,42 @@ shaka.util.ObjectUtils = class {
     return clone;
   }
 
+  /**
+   * Performs a shallow comparison between two objects or arrays.
+   *
+   * @template T
+   * @param {T} a
+   * @param {T} b
+   * @return {boolean}
+   */
+  static shallowCompare(a, b) {
+    if (Object.is(a, b)) {
+      return true;
+    }
+    if (typeof a !== 'object' || b === null ||
+        typeof a !== 'object' || b === null) {
+      return false;
+    }
+    if (Object.getPrototypeOf(a) !== Object.getPrototypeOf(b)) {
+      return false;
+    }
+    const keysA = Object.keys(a);
+    const keysB = Object.keys(b);
+    if (keysA.length !== keysB.length) {
+      return false;
+    }
+    for (let i = 0; i < keysA.length; i++) {
+      if (
+        // eslint-disable-next-line no-restricted-syntax
+        !Object.prototype.hasOwnProperty.call(b, keysA[i]) ||
+        !Object.is(a[keysA[i]], b[keysA[i]])
+      ) {
```

Comment:
Do we really care is it own property or inherited?
```suggestion
      if (!Object.is(a[keysA[i], b[keysA[i])) {
```